### PR TITLE
Form variables

### LINKED ISSUE #267: Rethinking what a variable is in Bitters
I’d like to investigate a refinement to how we set up Sass variables in Bitters.

## Problem

Right now, we have one `_variables.scss` partial, and every Sass variable in Bitters is defined there. This is fine in the context of Bitters; the project itself. But when a project that uses Bitters grows and variables are added, I’ve found the `_variables.scss` partial often balloons to upwards of 100 lines (highest I’ve seen yet is 377!). It makes it very hard to parse and feels unorganized to me. You begin to see things like this:

```scss
// Core Bitters variables above

// Badges
$badge-size: 1em;

// Tooltips
$tooltip-background-color: $dark-gray;

// Component foo
$component-foo-variable: value;

// etc.
```

While technically these variables are global (because Sass), I’d bet each of them are only used in one place: `components/_badges.scss` and `components/_tooltips.scss`, respectively (or something similar).

I’ve been asking myself a lot lately: Why define these in a partial with global project variables and not alongside the component styles they are strictly used in?

## Possible solution

Couple thoughts here…

Variables are a Sass _feature_. We’ve named this partial after the feature, and so I think people feel encouraged to put _all_ variables here, when that might not best solve for the job-to-be-done.

**We could rename `_variables.scss` to `_global-tokens.scss`**. I’d be open to other names, but quite a few people have started using “token” as a term to describe global, project-specific _decisions_ that establish a design system:

- https://www.lightningdesignsystem.com/design-tokens/
- https://medium.com/eightshapes-llc/tokens-in-design-systems-25dd82d58421#.nim8gy38q
- https://github.com/material-foundation/material-remixer (Google does use the term “ variable” here, but idea of them being extracted and separate from the component level is similar)

Encourage (unsure how Bitters could specifically help with this) component-level variables to be defined within the component’s partial. Here‘s an example showing this, and how it ties back to global tokens:

```scss
// _tokens.scss

// Colors
$medium-gray: #767676;
$yellow: #fff6bf;
$red: #fbe3e4;
```

```scss
// _button.scss

$_button-default-background-color: $medium-gray;

.c-button {
  background-color: $_button-default-background-color;
}
```

```scss
// _flashes.scss

$_flash-color-alert: #fff6bf;
$_flash-color-error: #fbe3e4;

.flash--alert {
  background-color: $_flash-color-alert;
}

.flash--error {
  background-color: $_flash-color-error;
}
```

_Note_: The underscore in `$_button-default-background-color` denotes it is a partial used only within the partial in which it is defined in.

The immediate change in Bitters with this idea would be to put the [`$form-` variables](https://github.com/thoughtbot/bitters/blob/master/core/_variables.scss#L40-L42) inside the `_forms` partial.

Thoughts?

### PR DESCRIPTION
**Pull request by tysongach**
- Put component-specific variables in its partial
  - Component-specific variables are variables that are used only for one component. They are unique from global variables in this way.
  - They should be prefixed by one underscore to indicate that they are “private” to the component/partial they are defined within.
  - Closes #267
 - Simplify input focus box-shadow

### CODE REVIEW

**Comment by creuter** [comment]
Is `form` a component if it's just a collection of default elements?

File: core/_forms.scss [hunk 1]
```diff
@@ -1,3 +1,6 @@
+$_form-box-shadow: inset 0 1px 3px rgba(#000, 0.06);
```
Comment (creuter) [inline core/_forms.scss]: Should we be aiming to squash all these? It's frustrating to have a bunch of Hound comments on a `bitters install`.

**Comment by tysongach** [comment]
> Is `form` a component if it's just a collection of default elements?

Yeah, I really meant component/object/element/partial…

File: core/_forms.scss (hunk 1 above)
Comment (tysongach) [inline core/_forms.scss]: It’d be nice to not have that happen, yeah. Do you want to open a PR for https://github.com/thoughtbot/bitters/issues/260?

**Comment by whmii** [comment]
lgtm
